app/controllers/reservations_controller.rb

```diff
@@ -24,11 +29,19 @@ def create
   end
 
   def show
+    authorize reservation
     render :show, locals: { reservation: reservation }
   end
 
-  def update
-    Reservations::UseCases::Update.new(reservation: reservation, status: params[:status]).call
+  def confirm
```

Comment:
It seems that your implementation allows confirming already cancelled reservations. Is it expected behaviour?

Please notice that your `cancel_reservation` implementation destroys all tickets connected with the reservation... so confirmation after `cancel_reservation` probably won't work as you expect...

Reply:
You're right, it's not expected behaviour. Thanks for pointing that out